pf/internal/check/check_test.go

```diff
@@ -96,6 +154,80 @@ func TestSensitiveIDWithOverride(t *testing.T) {
 	})
 }
 
+func TestInvalidInputID(t *testing.T) {
+	tests := []struct {
+		name          string
+		idSchema      idSchema
+		expectedError bool
+	}{
+		{name: "Required", idSchema: idSchema{required: true, optional: false, computed: false}, expectedError: true},
+		{name: "Optional", idSchema: idSchema{required: false, optional: true, computed: false}, expectedError: true},
+		{name: "Computed", idSchema: idSchema{required: false, optional: false, computed: true}, expectedError: false},
+		{name: "Optional+Computed", idSchema: idSchema{required: false, optional: true, computed: true}, expectedError: true},
+	}
+
+	for _, tc := range tests {
+		tc := tc
+		t.Run(tc.name+"true", func(t *testing.T) {
+			stderr, err := test(t, tfbridge.ProviderInfo{
+				P: pfbridge.ShimProvider(testProvider{withID: &idSchema{required: true}}),
+			})
+			assert.NotEmpty(t, stderr)
+			assert.Error(t, err)
+		})
+		t.Run(tc.name+"true (missing ComputeID)", func(t *testing.T) {
+			stderr, err := test(t, tfbridge.ProviderInfo{
+				P: pfbridge.ShimProvider(testProvider{withID: &tc.idSchema}),
+				Resources: map[string]*tfbridge.ResourceInfo{
+					"test_res": {Fields: map[string]*tfbridge.SchemaInfo{
+						"id": {Name: "otherId"},
+					}},
+				},
+			})
+			if tc.expectedError {
```

Comment:
Does the no-error case here actually work end to end? I wonder if the bridge treats id as non-identity otherId field, does it choke at runtime when trying to pick the identity? Like should we be insisting on ComputeID here?

Reply:
Yes it does fail at runtime with an error `Could not find required property 'otherId' in state`. I added some other runtime tests to check some of these cases.

Reply:
Done